### PR TITLE
Sorting columns (Actual and Initial value) issues solved; fixes #92

### LINKED ISSUE #92: Impossible to sort on columns in the OPC object browser
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the DEHP-EcosimPro
- [x] I have searched [open](https://github.com/RHEAGROUP/DEHP-ECOSIMPRO/issues) and [closed](https://github.com/RHEAGROUP/DEHP-ECOSIMPRO/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

when trying to order the OPC variables by either Actual or Initial Value using the table Headers, a InvalidOperationException is thrown.
There is a mismatch in the column types.
### System Configuration

- DEHPCommon version:
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

### PR DESCRIPTION
**Pull request by antoineatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEHP-ECOSIMPRO/pulls) open
- [x] I have verified that I am following the DEHP-EcosimPro [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEHP-ECOSIMPRO/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixes #92 
* Add the conversion to double when the data.Value was an int
* The conversion solve the null value of certain averaged value
* On OPC array rows, moved the dimensions of the array to the name

### CODE REVIEW

**Review by alexatrhea — CHANGES_REQUESTED** [review]
Unless I'm mistaken you should be able to handle more than just doubles.

File: DEHPEcosimPro/Behaviors/TreeListSortingBehavior.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+namespace DEHPEcosimPro.Behaviors
+{
+    using System;
+
+    using DevExpress.Mvvm.UI.Interactivity;
+    using DevExpress.Xpf.Grid;
+    using DevExpress.Xpf.Grid.TreeList;
+
+    public class TreeListSortingBehavior: Behavior<TreeListView>
```
Comment (alexatrhea) [inline DEHPEcosimPro/Behaviors/TreeListSortingBehavior.cs:33]: comments missing

File: DEHPEcosimPro.Tests/Behaviors/TreeListSortingBehaviorTestFixture.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="TreeListSortingBehaviorTestFixture.cs" company="RHEA System S.A.">
+//    Copyright (c) 2020-2022 RHEA System S.A.
+// 
+//    Author: Sam Gerené, Alex Vorobiev, Alexander van Delft, Nathanael Smiechowski.
```
Comment (alexatrhea) [inline DEHPEcosimPro.Tests/Behaviors/TreeListSortingBehaviorTestFixture.cs:5]: Please add your name to the headers

File: DEHPEcosimPro/Behaviors/TreeListSortingBehavior.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        }
+
+        /// <summary>
+        /// Handle the sort behaviour as we have to compare non-specific object types
+        /// </summary>
+        /// <param name="sender">The event sender</param>
+        /// <param name="e">The argument</param>
+        public void AssociatedObject_CustomColumnSort(object sender, TreeListCustomColumnSortEventArgs e)
+        {
+            IComparable value1 = e.Value1 as IComparable;
```
Comment (alexatrhea) [inline DEHPEcosimPro/Behaviors/TreeListSortingBehavior.cs:60]: in this entire method you should use `var` instead of explicit declaration.

**Review by alexatrhea — CHANGES_REQUESTED** [review]
Few small changes requested, Good approach with the Behavior!

**Comment by alexatrhea** [comment]
Also seems that coverage is not there yet.

**Review by samatrhea — APPROVED** [review]
looks good to me!

**Review by alexatrhea — APPROVED** [review]
Approved

**Review by nathanatrhea — APPROVED** [review]

**Pull request merged**